AG-10525 Fix tooltip positioning

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-10525

The bug was happening only when the enterprise package was used. It happened on line/scatter series when error bars were disabled.

This happened because ErrorBars.getDatum was adding an `undefined` and number value together, which apparently results in NaN. So the module was creating error bars with NaN translations.

To safe-guard against this, ErrorBars.getDatum now explicitly returns `number | undefined` types instead of `any` types.

Additionally, ErrorBars.createNodeData() has been modified to avoid creating error bar node-data when error bars are disabled.

olegat committed:
AG-10525 Fix tooltip positioning

The bug was happening only when the enterprise package was used. It
happened on line/scatter series when error bars were disabled.

This happened because ErrorBars.getDatum was adding an `undefined` and
number value together, which apparently results in NaN. So the module was
creating error bars with NaN translations.

To safe-guard against this, ErrorBars.getDatum now explicitly returns
`number | undefined` types instead of `any` types.

Additionally, ErrorBars.createNodeData() has been modified to avoid
creating error bar node-data when error bars are disabled.

alantreadway commented on packages/ag-charts-enterprise/src/features/error-bar/errorBar.ts:
```diff
@@ -242,6 +247,27 @@ export class ErrorBars extends _ModuleSupport.BaseModuleInstance implements _Mod
+        // The datum has an error value for `key`. Validate this user input value:
+        if (typeof value !== 'number') {
+            _Util.Logger.warnOnce(
+                `Found [${key}] error value (${value}) of type ${typeof value}. Expected number type`
```
Nit: I would not include the raw `value` in the message, as the deduplication effect of `warnOnce` is likely to be negated by different non-numeric numbers? (`typeof value` seems sufficient)

olegat replied:
Done. I've removed the `(${value})` part of the warning message.

alantreadway reviewed (commented):
LGTM aside from one nit

olegat committed:
AG-10525 CR: Don't expand ${value} in warning log

This is to avoid the deduplication effect of `warnOnce` to be negated
by different values, which defeats the purpose of using `warnOnce`.

alantreadway reviewed (approved):
LGTM
